projects/kit/components/slider/helpers/slider-thumb-label/slider-thumb-label.component.ts

```diff
@@ -0,0 +1,57 @@
+import {
+    AfterContentInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChild,
+    Inject,
+} from '@angular/core';
+import {tuiAssert, TuiDestroyService, tuiWatch} from '@taiga-ui/cdk';
+import {TuiSizeS} from '@taiga-ui/core';
+import {Observable} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+import {TuiSliderComponent} from '../../slider.component';
+
+@Component({
+    selector: '[tuiSliderThumbLabel]',
+    templateUrl: './slider-thumb-label.template.html',
+    styleUrls: ['./slider-thumb-label.style.less'],
+    providers: [TuiDestroyService],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiSliderThumbLabelComponent implements AfterContentInit {
+    @ContentChild(TuiSliderComponent)
+    readonly slider?: TuiSliderComponent;
+
+    get size(): TuiSizeS {
+        return this.slider?.size || 'm';
+    }
+
+    get ratio(): number {
+        return (this.slider?.valuePercentage || 0) / 100;
+    }
+
+    get ghostLeft(): number {
+        return this.ratio * (this.slider?.elementRef.nativeElement.offsetWidth || 0);
+    }
+
+    constructor(
+        @Inject(ChangeDetectorRef) private readonly changeDetectorRef: ChangeDetectorRef,
+        @Inject(TuiDestroyService) private readonly destroy$: Observable<unknown>,
+    ) {}
+
+    ngAfterContentInit(): void {
+        if (this.slider?.control?.valueChanges) {
```

Comment:
nit: how about `<ng-container *ngIf="slider?.control?.valueChanges | async"></ng-container>` in template? 
seems like we can do it without constructor/cdr/destroy
(assertion is convenient but i think we can sacrifice it, because it's obvious that label required slider)

Reply:
Good idea!
I've removed explicit subscription inside the component ✅

But about assertion.
It checks that the developer uses slider with angular forms (there is also option to use native attribute `value`).
Without Angular Forms it will work correctly (change detection problem).
Watch `input`-event is not good idea too, because it is not triggered if value was programatically patched. That is why angular forms support only for this helper (i think it is enough for developers).